crates/uv-workspace/src/workspace.rs

```diff
@@ -239,53 +239,93 @@ impl Workspace {
         }
     }
 
-    /// Returns the set of requirements that include all packages in the workspace.
-    pub fn members_as_requirements(&self) -> Vec<Requirement> {
-        self.packages
+    pub fn is_virtual(&self) -> bool {
+        !self
+            .packages
             .values()
-            .filter_map(|member| {
-                let project = member.pyproject_toml.project.as_ref()?;
-                // Extract the extras available in the project.
-                let extras = project
-                    .optional_dependencies
-                    .as_ref()
-                    .map(|optional_dependencies| {
-                        // It's a `BTreeMap` so the keys are sorted.
-                        optional_dependencies
-                            .iter()
-                            .filter_map(|(name, dependencies)| {
-                                if dependencies.is_empty() {
-                                    None
-                                } else {
-                                    Some(name)
-                                }
-                            })
-                            .cloned()
-                            .collect::<Vec<_>>()
-                    })
-                    .unwrap_or_default();
-
-                let url = VerbatimUrl::from_path(&member.root)
-                    .expect("path is valid URL")
-                    .with_given(member.root.to_string_lossy());
-                Some(Requirement {
-                    name: project.name.clone(),
-                    extras,
-                    marker: None,
-                    source: RequirementSource::Directory {
-                        install_path: member.root.clone(),
-                        lock_path: member
-                            .root
-                            .strip_prefix(&self.install_path)
-                            .expect("Project must be below workspace root")
-                            .to_path_buf(),
-                        editable: true,
-                        url,
-                    },
-                    origin: None,
+            .any(|member| *member.root() == self.install_path)
+    }
+
+    /// Returns the set of requirements that include all packages in the workspace.
+    pub fn members_as_requirements(&self) -> impl Iterator<Item = Requirement> + '_ {
```

Comment:
Minor preference for just `members_requirements` and `root_requirements` the `as` is a little confusing.